ecommerce/frontend/src/components/checkout/EditCardModal.js

```diff
@@ -0,0 +1,157 @@
+import { useState } from 'react'
+import Cards from 'react-credit-cards'
+import 'react-credit-cards/es/styles-compiled.css'
+import { Modal, Form, Input } from 'antd'
+
+export const EditCardModal = ({
+    cardProps = {
+        cvc: '',
+        expiry: '',
+        number: '',
+        name: '',
+    },
+    cardId = {},
+    cardName = '',
+    visible = false,
+    setVisible = () => {},
+}) => {
```

Comment:
Instead of having EditCardModal and AddCardModal have a single parent component that takes the id of the card and the "editing" mode as "add" or "edit" and then pass down the initial value (if there is) and an onSubmit callback to process the new values. Also instead of taking setVisibile take onOk and onCancel instead.